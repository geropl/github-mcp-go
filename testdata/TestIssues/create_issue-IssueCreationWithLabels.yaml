---
version: 2
interactions:
    - id: 0
      request:
        proto: HTTP/1.1
        proto_major: 1
        proto_minor: 1
        content_length: 135
        transfer_encoding: []
        trailer: {}
        host: api.github.com
        remote_addr: ""
        request_uri: ""
        body: |
            {"title":"Test Issue with Labels","body":"This is a test issue with labels created by the test suite.","labels":["bug","enhancement"]}
        form: {}
        headers:
            Accept:
                - application/vnd.github.v3+json
            Content-Type:
                - application/json
            User-Agent:
                - go-github/v69.2.0
            X-Github-Api-Version:
                - "2022-11-28"
        url: https://api.github.com/repos/geropl/github-mcp-go-test/issues
        method: POST
      response:
        proto: HTTP/2.0
        proto_major: 2
        proto_minor: 0
        transfer_encoding: []
        trailer: {}
        content_length: 2677
        uncompressed: false
        body: '{"url":"https://api.github.com/repos/geropl/github-mcp-go-test/issues/16","repository_url":"https://api.github.com/repos/geropl/github-mcp-go-test","labels_url":"https://api.github.com/repos/geropl/github-mcp-go-test/issues/16/labels{/name}","comments_url":"https://api.github.com/repos/geropl/github-mcp-go-test/issues/16/comments","events_url":"https://api.github.com/repos/geropl/github-mcp-go-test/issues/16/events","html_url":"https://github.com/geropl/github-mcp-go-test/issues/16","id":2902874257,"node_id":"I_kwDOOEmhcs6tBliR","number":16,"title":"Test Issue with Labels","user":{"login":"geropl","id":32448529,"node_id":"MDQ6VXNlcjMyNDQ4NTI5","avatar_url":"https://avatars.githubusercontent.com/u/32448529?v=4","gravatar_id":"","url":"https://api.github.com/users/geropl","html_url":"https://github.com/geropl","followers_url":"https://api.github.com/users/geropl/followers","following_url":"https://api.github.com/users/geropl/following{/other_user}","gists_url":"https://api.github.com/users/geropl/gists{/gist_id}","starred_url":"https://api.github.com/users/geropl/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/geropl/subscriptions","organizations_url":"https://api.github.com/users/geropl/orgs","repos_url":"https://api.github.com/users/geropl/repos","events_url":"https://api.github.com/users/geropl/events{/privacy}","received_events_url":"https://api.github.com/users/geropl/received_events","type":"User","user_view_type":"public","site_admin":false},"labels":[{"id":8255618595,"node_id":"LA_kwDOOEmhcs8AAAAB7BK-Iw","url":"https://api.github.com/repos/geropl/github-mcp-go-test/labels/bug","name":"bug","color":"d73a4a","default":true,"description":"Something isn''t working"},{"id":8255618610,"node_id":"LA_kwDOOEmhcs8AAAAB7BK-Mg","url":"https://api.github.com/repos/geropl/github-mcp-go-test/labels/enhancement","name":"enhancement","color":"a2eeef","default":true,"description":"New feature or request"}],"state":"open","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":0,"created_at":"2025-03-07T12:24:13Z","updated_at":"2025-03-07T12:24:13Z","closed_at":null,"author_association":"OWNER","sub_issues_summary":{"total":0,"completed":0,"percent_completed":0},"active_lock_reason":null,"body":"This is a test issue with labels created by the test suite.","closed_by":null,"reactions":{"url":"https://api.github.com/repos/geropl/github-mcp-go-test/issues/16/reactions","total_count":0,"+1":0,"-1":0,"laugh":0,"hooray":0,"confused":0,"heart":0,"rocket":0,"eyes":0},"timeline_url":"https://api.github.com/repos/geropl/github-mcp-go-test/issues/16/timeline","performed_via_github_app":null,"state_reason":null}'
        headers:
            Access-Control-Allow-Origin:
                - '*'
            Access-Control-Expose-Headers:
                - ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset
            Cache-Control:
                - private, max-age=60, s-maxage=60
            Content-Length:
                - "2677"
            Content-Security-Policy:
                - default-src 'none'
            Content-Type:
                - application/json; charset=utf-8
            Date:
                - Fri, 07 Mar 2025 12:24:14 GMT
            Etag:
                - '"95d8b0143dd9e600ac23c79dc3273f7edaa63520bc2d098135217dac8e03a0b6"'
            Github-Authentication-Token-Expiration:
                - 2025-04-06 13:14:03 +0200
            Location:
                - https://api.github.com/repos/geropl/github-mcp-go-test/issues/16
            Referrer-Policy:
                - origin-when-cross-origin, strict-origin-when-cross-origin
            Server:
                - github.com
            Strict-Transport-Security:
                - max-age=31536000; includeSubdomains; preload
            Vary:
                - Accept, Authorization, Cookie, X-GitHub-OTP,Accept-Encoding, Accept, X-Requested-With
            X-Accepted-Github-Permissions:
                - issues=write
            X-Content-Type-Options:
                - nosniff
            X-Frame-Options:
                - deny
            X-Github-Api-Version-Selected:
                - "2022-11-28"
            X-Github-Media-Type:
                - github.v3; format=json
            X-Github-Request-Id:
                - FE43:3947F4:3A22309:3BFB0AC:67CAE56D
            X-Ratelimit-Limit:
                - "5000"
            X-Ratelimit-Remaining:
                - "4978"
            X-Ratelimit-Reset:
                - "1741353443"
            X-Ratelimit-Resource:
                - core
            X-Ratelimit-Used:
                - "22"
            X-Xss-Protection:
                - "0"
        status: 201 Created
        code: 201
        duration: 654.294916ms
