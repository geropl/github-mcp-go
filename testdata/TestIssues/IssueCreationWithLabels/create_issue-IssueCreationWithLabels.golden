{
  "output": "# Issue: Test Issue with Labels\n\n**Number:** #16  \n**State:** open  \n**Created:** Fri, 07 Mar 2025 12:24:13 UTC  \n**URL:** https://github.com/geropl/github-mcp-go-test/issues/16  \n\n## Description\n\nThis is a test issue with labels created by the test suite.\n\n## Details\n\n**Labels:**  \n- bug  \n- enhancement  \n\n**Comments:** 0  \n",
  "err": ""
}